android/java/org/chromium/chrome/browser/settings/NetworkPreferenceAdapter.java

```diff
@@ -0,0 +1,544 @@
+                    () -> {
+                        View contentView =
+                                LayoutInflater.from(mContext)
+                                        .inflate(
+                                                R.layout.app_menu_layout,
+                                                networkViewHolder.mItem,
+                                                false);
+                        ListView listView = contentView.findViewById(R.id.app_menu_list);
+                        return new BasicListMenu(
+                                mContext, menuItems, contentView, listView, delegate, 0);
+                    });
+        } else {
+            final LabelViewHolder labelViewHolder = (LabelViewHolder) viewHolder;
+            labelViewHolder.mLabel.setText(networkPreferenceItem.getNetworkNameRes());
+        }
+    }
+
+    private boolean isHiddenNetwork(@NonNull final NetworkInfo network) {
+        return (network.coin == CoinType.ETH && mHiddenEthChainIds.contains(network.chainId))
+                || (network.coin == CoinType.FIL && mHiddenFilChainIds.contains(network.chainId))
+                || (network.coin == CoinType.SOL && mHiddenSolChainIds.contains(network.chainId))
+                || (network.coin == CoinType.BTC && mHiddenBtcChainIds.contains(network.chainId));
+    }
+
+    private boolean isActiveNetwork(@NonNull final NetworkInfo network) {
+        return (network.coin == CoinType.ETH && network.chainId.equals(mActiveEthChainId))
+                || (network.coin == CoinType.FIL && network.chainId.equals(mActiveFilChainId))
+                || (network.coin == CoinType.SOL && network.chainId.equals(mActiveSolChainId))
+                || (network.coin == CoinType.BTC && network.chainId.equals(mActiveBtcChainId));
+    }
+
+    private int findChainIdPosition(
+            @NonNull final String chainId, @CoinType.EnumType final int coinType) {
+        for (int i = 0; i < mElements.size(); i++) {
+            NetworkPreferenceItem networkPreferenceItem = mElements.get(i);
+            if (networkPreferenceItem.mNetworkInfo != null
+                    && chainId.equals(networkPreferenceItem.mNetworkInfo.chainId)
+                    && coinType == networkPreferenceItem.mNetworkInfo.coin) {
+                return i;
+            }
+        }
+        return NO_POSITION;
+    }
+
+    @NonNull
+    private String getActiveChainId(@CoinType.EnumType final int coinType) {
+        return switch (coinType) {
+            case CoinType.ETH -> mActiveEthChainId;
+            case CoinType.FIL -> mActiveFilChainId;
+            case CoinType.SOL -> mActiveSolChainId;
+            case CoinType.BTC -> mActiveBtcChainId;
+            case CoinType.ZEC -> throw new IllegalStateException(
+                    String.format("Unsupported active chain for coin type %d.", coinType));
+            default -> throw new IllegalStateException("Unexpected value: " + coinType);
```

Comment:
it would be good to wrap all calls to `getActiveChainId`, `setActiveChainId` and `addHiddenNetwork`, `removeHiddenNetwork` with try catch

Reply:
Try/catch affect the performance and I'd prefer to use them as a last resort. 
I'm going to implement a better solution that does not include throwing an exception.